### ISSUE TITLE
Make version_1.4.1-dev1 a full release

### ISSUE THREAD

**Issue by teleksterling** [markhannon/zimi-integration#19]
Hey @markhannon , 1.4.1-dev1 has been running smoothly since Feb, and it will help address another user's issue recently discussed in the Zimi forums thread.

Think it can graduate from pre-release?

**Comment by markhannon** [comment]
Hi Rick, 
Good to hear it has been stable - I have been working with the HA team to try and get the code actually included in the HA core and as part of that I released a new zcc helper version 3.2.2 which I am testing within the HA core integration.   I have just bundled that up into a new pre-release version of the HACS repo.   Please upgrade and test.   If it is stable for you then I will promote that to full release

**Comment by teleksterling** [comment]
I'll do it now!

**Comment by teleksterling** [comment]
Hmm, HACS says:

Release version_1.4.1-rc1
The version version_1.4.1-rc1 for this integration can not be used with HACS.

Perhaps it's part of the new caching system? It might work tomorrow.

**Comment by sachitgagneja2530** [comment]
Thanks @teleksterling @markhannon for taking this up.
I tried the previous pre-release version weeks ago. Was getting the same error. And same is happening with this version as well. 
<img width="1110" alt="Screenshot 2024-11-10 at 3 14 22 PM" src="https://github.com/user-attachments/assets/6311d030-0898-4ef4-9bf8-7568536094c0">

**Comment by markhannon** [comment]
OK, I renumbered the version to 1.4.1 and flagged as pre-release.   Please check and confirm it (a) loads via HACS and (b) behaves

**Comment by sachitgagneja2530** [comment]
No, cant load it via HACS. Still the same error. Not sure if it makes any difference, but I am trying to upgrade from current 1.4.0 version. 
<img width="1016" alt="Screenshot 2024-11-10 at 4 35 24 PM" src="https://github.com/user-attachments/assets/9748e20f-6600-48bf-9930-21af83f2d1af">

**Comment by teleksterling** [comment]
Install issue identified - see #20

**Comment by markhannon** [comment]
Thanks Rick for chasing down.   Looks like I need to create a new file!   Can you create one with a PR?

**Comment by teleksterling** [comment]
Sure thing - later tonight

**Comment by markhannon** [comment]
@sachitgagneja2530 @teleksterling  Please check version 1.4.2.   It is marked as a pre-release and I added the hacs.json file

**Comment by sachitgagneja2530** [comment]
Thanks guys! It worked perfectly. No issues at all.

**Comment by teleksterling** [comment]
Same here - I've updated to version 1.4.2 without issue, and it's running well. 

Interesting to note though, while I was out version 1.4.1 did somehow install/update - i.e. prior to hacs.json being added. So although the missing file caused an error, it didn't stop the installation given enough time. 

I still wonder if it's something to do with HACS caching version checks. Perhaps when they refresh their cache they also add some metadata to account for repos with missing files? 

In any case this repo is more compliant now, and I guess this issue ought to wait until 1.4.2 (or .3) is released to be closed. 

Thanks @markhannon !!

**Comment by markhannon** [comment]
I was planning on making a full release on the weekend but now it seem OP still has issues on 1.4.2

Regards, Mark
________________________________

**Comment by sachitgagneja2530** [comment]
Hey @markhannon @teleksterling, not sure if its happening with you but there seems to be an issue with version 1.4.2 where the garage door status takes forever to update. When I open the door, the status doesnt change for a while - sometimes it takes up to 30 minutes. Because of that the close button stays greyed out so I cant close the door right away.

Its also causing problems with automations. Since the status isnt updating, opening or closing the garage door doesnt trigger the rest of the automation flow. I hope its not just me !

**Comment by markhannon** [comment]
Not sure if it is just you.   The main garage user has been Rick and I believe he is stable now.

Can you enable debug on the integration and send some logs please?   Please note in the logs when you started the action.    You should see a whole series of updates from the ZCC as it moves the door bit by bit.   If they are not happening we may have an issue in the zimi backend.

I don’t have a garage door to test with.

From: sachitgagneja2530 ***@***.***>
Date: Friday, 15 November 2024 at 12:44 pm
To: markhannon/zimi-integration ***@***.***>
Cc: markhannon ***@***.***>, Mention ***@***.***>
Subject: Re: [markhannon/zimi-integration] Make version_1.4.1-dev1 a full release (Issue #19)

Hey @markhannon<https://github.com/markhannon> @teleksterling<https://github.com/teleksterling>, not sure if its happening with you but there seems to be an issue with version 1.4.2 where the garage door status takes forever to update. When I open the door, the status doesnt change for a while - sometimes it takes up to 30 minutes. Because of that the close button stays greyed out so I cant close the door right away.

Its also causing problems with automations. Since the status isnt updating, opening or closing the garage door doesnt trigger the rest of the automation flow. I hope its not just me !

**Comment by teleksterling** [comment]
I'm having issues with 1.4.2 now too - the whole config won't load. But the debug log doesn't show any obvious show stoppers. It coincided with be updating to HA 2024.11.2. But I've since rolled back to 2024.11.1 and have the same issue. 

Also tested zimi 1.4.0 which loads fine, but obviously no garage controls. 

I'll compile my logs and attach later on.

**Comment by markhannon** [comment]
Wow.   Sorry to hear that.   I am away for a few days and will have a look on the weekend at the logs

Regards, Mark
________________________________

**Comment by markhannon** [comment]
Hi Rick, Can you capture and upload some logs with debug set for the integration please?   I have just asked the zimi guys if anything changed, but maybe it is a HA thing.   The garage door integration is different from all the others in that it sends a continuous stream of update messages rather than on/off status updates.    Might be something in the HA side or my zcc library that is now causing an issue

**Comment by Picco-10** [comment]
Hi Guys, My garage door also stopped working once HA was upgraded to 11.2 & 11.3. i was using zimi 1.41 i have since upgraded to 1.42 still no luck. i have been away so havent had much time. i have attached some logs. let me know if i can help any further
thanks 
[home-assistant_zimi_2024-11-27T01-42-24.964Z.log](https://github.com/user-attachments/files/17927657/home-assistant_zimi_2024-11-27T01-42-24.964Z.log)

**Comment by markhannon** [comment]
Sorry to hear that.   Can you please change the verbosity to 3 and retry to open and close a garage door please?   That way I can see the API messages back and forwards from the zcc and hopefully debug this.

________________________________

**Comment by markhannon** [comment]
@ABdataman Have you tried your garage lately?   Can you also set verbosity high and test open and close please?

**Comment by Picco-10** [comment]
@markhannon here is the new log. hopefully thats right. 
[home-assistant_zimi_2024-11-27T03-10-24.056Z.log](https://github.com/user-attachments/files/17928287/home-assistant_zimi_2024-11-27T03-10-24.056Z.log)

**Comment by markhannon** [comment]
Unfortunately that doesn't have the verbose API messages.    These will only be captured if the integration is configured with verbosity=3.

You can either delete and then add the zcc again (should not have any issues with re-discovering existing devices with no changes unless you have renamed them in HA) with verbosity=3.

I will need to re-look at how this is done for the official core integration I am working on, HA frowns on too much debug.

________________________________

**Comment by Picco-10** [comment]
@markhannon here is the updated version. deleted and re-added with Verbosity =3
[home-assistant_zimi_2024-11-27T04-27-49.977Z.log](https://github.com/user-attachments/files/17928949/home-assistant_zimi_2024-11-27T04-27-49.977Z.log)

**Comment by teleksterling** [comment]
Hi. 
Just a quick update - I discovered midweek last week that the Zimi app didn't have any open/close history since the Monday.

The would have coincided with when I started having the issues with the integration. 

I cut power to the zcc and garage controller and everything came back to life. 

I don't know whether something about 1.4.2 triggered the system to lock up, or something else.

**Comment by sachitgagneja2530** [comment]
Still having issues with the current version. Here are the logs. Garage opens fine but cant stop it in between and cant close it. The close button is always greyed out.

[home-assistant_zimi_2024-12-17T12-53-28.120Z.log](https://github.com/user-attachments/files/18165786/home-assistant_zimi_2024-12-17T12-53-28.120Z.log)
[home-assistant_zimi_202

https://github.com/user-attachments/assets/4d7fbf74-eed8-4c21-9ac6-c3866ef68a68

4-11-27T04-27-49.977Z.log](https://github.com/user-attachments/files/18165788/home-assistant_zimi_2024-11-27T04-27-49.977Z.log)

**Comment by markhannon** [comment]
Do you use HA to operate the other zimi lights and switches?

I can see that after you try to open the door the controller fails as it no
longer has a socket connection to the zcc.    This doesn't make any sense
to me why this would be related to cover.

Can we arrange a live debug session next Monday?   You will need to be home
to observe the door and setup a port forward to access.   If this is OK
please contact me via email ***@***.*** to arrange.

**Comment by sachitgagneja2530** [comment]
I only have the garage controller set up - no switches or lights. Sure we can do a debug session but your email address was hidden by GitHub. My email address is included at the end of the log.
[Zimi Log.pdf](https://github.com/user-attachments/files/18173930/Zimi.Log.pdf)

**Comment by markhannon** [comment]
Is this still the case in version 2.0.1?
